### ISSUE TITLE
Synapse hangs with "too many open files"

### ISSUE THREAD

**Issue by WGH-** [matrix-org/synapse#1723]
My Synapse server has been working for nine days, and today it stopped responding.

The CPU usage is stuck at 100%, and there're many errors regarding to open file limit, like:

`twisted: [PUT-1187882] Could not accept new connection (EMFILE)`

````
twisted: [PUT-1187882] Unhandled Error
Traceback (most recent call last):
  File "/home/synapse/.env/lib/python2.7/site-packages/synapse/app/homeserver.py", line 455, in in_thread
    
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/internet/base.py", line 1199, in run
    
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/internet/base.py", line 1208, in mainLoop
    
  File "/home/synapse/.env/lib/python2.7/site-packages/synapse/metrics/__init__.py", line 151, in f
    
--- <exception caught here> ---
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/internet/base.py", line 828, in runUntilCurrent
    
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/names/client.py", line 152, in maybeParseConfig
    
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/names/client.py", line 143, in _openFile
    
  File "/home/synapse/.env/lib/python2.7/site-packages/twisted/python/filepath.py", line 1014, in open
    
exceptions.IOError: [Errno 24] Too many open files: '/etc/resolv.conf'
````

[Here is lsof output](https://github.com/matrix-org/synapse/files/675404/lsof.txt). 192.168.0.2:http-alt is local synapse port, 192.168.0.1 is nginx upstream.

**Comment by Half-Shot** [comment]
Been having this too. Synapse is now on the ``matrix`` user on my server (got around to doing it) with it's own file limits which seems to help somewhat.

**Comment by ara4n** [comment]
A workaround for this landed in 0.18.6-rc1 in the form of #1725.  The root cause looks to be #1733.
